src/AzureExtension/DevBox/DevBoxInstance.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System.Diagnostics;
+using System.Text.Json;
+using AzureExtension.Contracts;
+using Microsoft.Windows.DevHome.SDK;
+using Windows.Foundation;
+
+namespace AzureExtension.DevBox;
+
+public class DevBoxInstance : IComputeSystem
+{
+    private readonly IDevBoxAuthService _authService;
+
+    public IDeveloperId? DevId
+    {
+        get; private set;
+    }
+
+    public string? Name
+    {
+        get; private set;
+    }
+
+    public string? Id
+    {
+        get; private set;
+    }
+
+    public string? ProjectName
+    {
+        get; private set;
+    }
+
+    public string? State
+    {
+        get; private set;
+    }
+
+    public string? CPU
+    {
+        get; private set;
+    }
+
+    public string? Memory
+    {
+        get; private set;
+    }
+
+    public string? WebURI
+    {
+        get; private set;
+    }
+
+    public string? BoxURI
+    {
+        get; private set;
+    }
+
+    public string? RdpURI
+    {
+        get; private set;
+    }
+
+    public string? OS
+    {
+        get; private set;
+    }
+
+    public DevBoxInstance(IDevBoxAuthService devBoxAuthService)
+    {
+        _authService = devBoxAuthService;
+    }
+
+    // Returns a DevBox object from a JSON object
+    public void FillFromJson(JsonElement item, string project, IDeveloperId? devId) => throw new NotImplementedException();
+
+    private (string WebURI, string RdpURI) GetRemoteLaunchURIsAsync(string boxURI)

```

Comment:
C# has System.Uri.  I suggest using that instead of strings.

Reply:
Will take care of in the next PR.